pkg/apis/settings/settings.go

```diff
@@ -31,15 +31,17 @@ type settingsKeyType struct{}
 var ContextKey = settingsKeyType{}
 
 var defaultSettings = &Settings{
-	BatchMaxDuration:  metav1.Duration{Duration: time.Second * 10},
-	BatchIdleDuration: metav1.Duration{Duration: time.Second * 1},
-	DriftEnabled:      false,
+	BatchMaxDuration:      &metav1.Duration{Duration: time.Second * 10},
+	BatchIdleDuration:     &metav1.Duration{Duration: time.Second * 1},
+	TTLAfterNotRegistered: &metav1.Duration{Duration: time.Minute * 15},
+	DriftEnabled:          false,
 }
 
 // +k8s:deepcopy-gen=true
 type Settings struct {
-	BatchMaxDuration  metav1.Duration
-	BatchIdleDuration metav1.Duration
+	BatchMaxDuration      *metav1.Duration `validate:"required"`
+	BatchIdleDuration     *metav1.Duration `validate:"required"`
```

Comment:
Default settings are set above, if these are null by someone setting them to an empty string, we will error out with a failed validation since these values are marked as required.